File: abc/abc155/abc155_b.cs
// https://atcoder.jp/contests/abc155/submissions/10162819

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var N = int.Parse(Console.ReadLine());
            var A = Console.ReadLine().Split().Select(int.Parse).ToArray();

            var flag = true;
            for (int i = 0; i < N; i++)
            {
                if (A[i] % 2 == 0)
                {
                    if (A[i] % 3 != 0 && A[i] % 5 != 0)
                    {
                        flag = false;
                    }
                }
            }
            Console.WriteLine(flag ? "APPROVED" : "DENIED");
        }
    }
}

File: abc/abc155/abc155_b.fs
// https://atcoder.jp/contests/abc155/submissions/11909200

let N = stdin.ReadLine()

stdin.ReadLine().Split()
|> Array.map int
|> Array.filter (fun x -> x % 2 = 0)
|> Array.filter (fun x -> x % 3 > 0)
|> Array.filter (fun x -> x % 5 > 0)
|> Array.isEmpty
|> fun x ->
    match x with
    | true -> "APPROVED"
    | false -> "DENIED"
|> stdout.WriteLine
